internal/pkg/manifest/workload_ecs.go

```diff
@@ -210,12 +210,20 @@ func (s *Secret) IsSecretsManagerName() bool {
 	return !s.fromSecretsManager.IsEmpty()
 }
 
+// SSMRequiresImport returns true if the SSM parameter name or secret ARN value is imported from CloudFormation stack.
+func (s *Secret) SSMRequiresImport() bool {
+	return !s.from.FromCFN.isEmpty()
+}
+
 // Value returns the secret value provided by clients.
 func (s *Secret) Value() string {
 	if !s.fromSecretsManager.IsEmpty() {
 		return aws.StringValue(s.fromSecretsManager.Name)
+	} else if !s.SSMRequiresImport() {
+		return aws.StringValue(s.from.Plain)
+	} else {
+		return aws.StringValue(s.from.FromCFN.Name)
```

Comment:
nit: can we flip the boolean so that the logic is more straightforward 
```suggestion
	} else if s.SSMRequiresImport() {
	        return aws.StringValue(s.from.FromCFN.Name)
	} else {
		return aws.StringValue(s.from.Plain)
```